### PR TITLE
[MRG] Improve gamma example by adding new `viz` methods

### PR DESCRIPTION
**Pull request by cjayb**
This PR

* improves the narrative of the gamma example
* adds `viz.plot_tfr_morlet`: utility for plotting TFR of single-trial dpl time course (uses MNE and SciPy)
* adds `viz.plot_spectrogram`: utility for plotting Welch periodogram of single-trial dpl time course (uses SciPy)

Visualisation functions are poorly-tested, and input validation is largely missing. However, the readability of the example has improved dramatically, IMHO.

### CODE REVIEW

**Comment by cjayb** [comment]
@jasmainak I don't get why the new `plot_tfr_morlet` and `plot_spectrogram` don't appear as hyperlinks in the examples? Some init file I need to include them in? They're in `api.rst`

@ntolley take a look at @2218c9c : you could use the new function in your alpha/beta demo?

For a 0.1-release, perhaps the missing parameter checks are an acceptable omission...?

File: hnn_core/viz.py
Comment (jasmainak) [inline hnn_core/viz.py:306]: freqs

File: examples/plot_simulate_gamma.py
Comment (jasmainak) [inline examples/plot_simulate_gamma.py:39]: Actually they shouldn't be dramatically different. Maybe only the layer2-layer5 connections are turned off and the rest is same? I'd like to move away from importing different param files to initialize the network. Do you think using `default.param` would work?

File: hnn_core/viz.py
Comment (jasmainak) [inline hnn_core/viz.py:343]: there seems to be a strong edge effect now!

File: examples/plot_simulate_gamma.py
Comment (cjayb) [reply examples/plot_simulate_gamma.py:39]: They are: just compare `param['gbar_L*]`. The paper (Law & Jones, 2013) is pretty clear about it too. I haven't tried using the default parameters, though, could be interesting...

File: hnn_core/viz.py
Comment (cjayb) [reply hnn_core/viz.py:343]: I noticed that, should I zero-pad instead?

Comment (jasmainak) [reply hnn_core/viz.py:343]: maybe give it a quick try? Let's not be too distracted by this

File: examples/plot_simulate_gamma.py
Comment (jasmainak) [reply examples/plot_simulate_gamma.py:39]: indeed ... let's leave it like this then

File: examples/plot_simulate_gamma.py
Comment (jasmainak) [inline examples/plot_simulate_gamma.py:76]: do we really need to do this? I think the fewer the parameters ... the better

**Review by jasmainak — APPROVED** [review]
Other than minor nitpicks, it's good to go @cjayb ! Thanks for improving the example

File: examples/plot_simulate_gamma.py
Comment (jasmainak) [inline examples/plot_simulate_gamma.py:14]: Maybe a heading "References" above this?

Comment (jasmainak) [reply examples/plot_simulate_gamma.py:14]: or alternatively add this to the end of the example?

File: hnn_core/viz.py
Comment (cjayb) [reply hnn_core/viz.py:343]: I've added zero-padding as the default, arriving in the next commit. Looks better than with mirroring for this dataset.

File: examples/plot_simulate_gamma.py
Comment (cjayb) [reply examples/plot_simulate_gamma.py:76]: My plotting is insanely slow without some serious decimating, can you test the interactive performance on yours?

**Comment by cjayb** [comment]
Thanks @jasmainak ! Any idea how I can get `plot_spectogram` and `plot_tfr_morlet` to be clickable in the example?

Screenshot 2021-01-21 at 20 02 50

**Comment by jasmainak** [comment]
> Any idea how I can get plot_spectogram and plot_tfr_morlet to be clickable in the example

I think it will be fixed once we merge this PR :-)

**Comment by jasmainak** [comment]
> /Users/mainak/anaconda3/lib/python3.7/site-packages/scipy/signal/spectral.py:1969: UserWarning: nperseg = 8192 is greater than input length  = 8000, using nperseg = 8000
  .format(nperseg, input_length))

I get this warning when running the example. Ideally, examples shouldn't throw warnings so user is not distracted.

**Comment by jasmainak** [comment]
Here's another warning I get:

```py
/Users/mainak/Documents/github_repos/hnn-core/hnn_core/viz.py:378: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
  cbar_ax = fig.add_axes([l + w + 0.05, cb_b, 0.03, cb_h])
```

File: examples/plot_simulate_gamma.py
Comment (jasmainak) [inline examples/plot_simulate_gamma.py:60]: so now the issue is that the user will be surprised with an import warning since the installation instructions didn't talk about installing MNE and neither does the example.

File: examples/plot_simulate_gamma.py
Comment (jasmainak) [reply examples/plot_simulate_gamma.py:71]: It's not "insanely slow" but yeah the `decim` does make it faster. I can live with it :)

```py
In [6]: %timeit plot_tfr_morlet(dpls[trial_idx], freqs=freqs, n_cycles=7, tmin=t
   ...: min, decim=8, ax=axes[1])                                               
/Users/mainak/Documents/github_repos/hnn-core/hnn_core/viz.py:378: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
  cbar_ax = fig.add_axes([l + w + 0.05, cb_b, 0.03, cb_h])
140 ms ± 5.53 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [7]: %timeit plot_tfr_morlet(dpls[trial_idx], freqs=freqs, n_cycles=7, tmin=t
   ...: min, decim=1, ax=axes[1])                                               
/Users/mainak/Documents/github_repos/hnn-core/hnn_core/viz.py:378: MatplotlibDeprecationWarning: Adding an axes using the same arguments as a previous axes currently reuses the earlier instance.  In a future version, a new instance will always be created and returned.  Meanwhile, this warning can be suppressed, and the future behavior ensured, by passing a unique label to each axes instance.
  cbar_ax = fig.add_axes([l + w + 0.05, cb_b, 0.03, cb_h])
940 ms ± 14.3 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

**Comment by cjayb** [comment]
I'll fix the MPL deprecation warning tomorrow and figure out a solution to the `nperseg` warning too (my version didn't throw that one...). As for `mne`-dependency... I agree we should probably remove deep ones like that, most likely copying relevant parts into `hnn_core`, as you mentioned. But perhaps acceptable for now (0.1)?

**Comment by jasmainak** [comment]
yes, we can do it after 0.1. My comment was mostly to say that let's add a line in the tutorial asking them to install `mne` for the particular example.

**Comment by jasmainak** [comment]
happy to merge once the warnings are fixed!

File: examples/plot_simulate_gamma.py
Comment (rythorpe) [reply examples/plot_simulate_gamma.py:71]: Mine ran with a reasonable execution time regardless of decimation as well, ~39 ms with decimation vs. ~295 ms without. For simplicity's sake, I'm not convinced we need an extra line defining the `decim` variable (just like we don't define a separate var for `n_cycles`) but that's stylistic.

**Review by rythorpe — APPROVED** [review]

**Comment by cjayb** [comment]
Alright, the last three commits should reduce the warnings and includes mention of `mne`. Please check that this is the case.

Note that the new `plt_show` method copied from `mne` should suppress warnings in all examples!

I'm done here!

**Comment by cjayb** [comment]
> Mine ran with a reasonable execution time regardless of decimation as well, ~39 ms with decimation vs. ~295 ms without. For simplicity's sake, I'm not convinced we need an extra line defining the decim variable (just like we don't define a separate var for n_cycles) but that's stylistic.

Ah, just noticed this. Maybe my macOS renderer is just inefficient. Feel free to remove the `decim`'s from the example before merging!

**Comment by jasmainak** [comment]
Thanks a ton @cjayb for these improvements!

**Pull request merged**
